pkg/kubelet/cm/topologymanager/scope_pod.go

```diff
@@ -57,11 +57,13 @@ func (s *podScope) Admit(pod *v1.Pod) lifecycle.PodAdmitResult {
 	for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
 		klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
 
+		s.Lock()
 		if (s.podTopologyHints)[string(pod.UID)] == nil {
 			(s.podTopologyHints)[string(pod.UID)] = make(map[string]TopologyHint)
 		}
 
 		(s.podTopologyHints)[string(pod.UID)][container.Name] = bestHint
+		s.Unlock()
```

Comment:
And this would become:
```
klog.Infof("[topologymanager] Topology Affinity for (pod: %v container: %v): %v", format.Pod(pod), container.Name, bestHint)
s.setTopologyHints(string(pod.UID), container.Name, bestHint)
```